include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -113,6 +113,12 @@ def GTWithConstLHS : Pat<
   (MoveNameHint $old, (LTPrimOp $rhs, $lhs)),
   [(AnyConstantOp $lhs), (NonConstantOp $rhs)]>;
 
+// not(not(x)) -> x
+def NotNot : Pat <
+  (NotPrimOp (NotPrimOp $x)),
+  (AsUIntPrimOp $x),
+  []>;
```

Comment:
Namehint ? 

:+1: for the `AsUIntPrimOp` in case input is signed! (which folds away if no-op, woo).

Reply:
But I just added a name to AsUIntPrimOp in another commit, so might as well.